### ISSUE TITLE
Attachments cannot be displayed!

### ISSUE THREAD

**Issue by alisey123** [Awesome-Support/Awesome-Support#354]
Attached images cannot be displayed from both front-end and back-end. 
The error message:

> The image "http://mywebsite.com/?wpas-attachment=14318" cannot be displayed because it contains errors.

**Comment by elindydotcom** [comment]
Can you give us some more info?  What is your default image viewer (Awesome Support does not have a native image viewer)?  What type of image?  Was it uploaded from a front end ticket or attached on the backend?    Thanks!

**Comment by alisey123** [comment]
I'm trying to view the images in chrome browser. 
Images are in jpg format
Images are uploaded from frontend when user was submitting a new ticket.

Thank you!

**Comment by elindydotcom** [comment]
hmmm....we  areunable to reproduce this.  Can you locate the attachment in /uploads/awesome-support/ticket_nnn (where nnn is the ticket number)?  If you download it and try to view it, does it work?

**Comment by alisey123** [comment]
When an image is attached, it shows up in wp library and in /uploads/awesome-support/ticket_nnn but the generated URL in the ticket does not open the image.
![untitled-1](https://cloud.githubusercontent.com/assets/3501211/21127131/e07015f8-c11a-11e6-81e8-ea0a8b18dda9.jpg)

Thank you!

**Comment by elindydotcom** [comment]
Can you locate the attachment in /uploads/awesome-support/ticket_nnn (where nnn is the ticket number)? If you download it and try to view it, does it work?

**Comment by alisey123** [comment]
yes images are available in /uploads/awesome-support/ticket_nnn and I can view if I download and open them!

**Comment by elindydotcom** [comment]
hmmm...is it happening on a clean install of WP (only Awesome Support installed and a core WP theme?)

**Comment by elindydotcom** [comment]
hmmmm...any chance you can spin up a clean version of WP and the latest dev branch of AS and let me know if you still see the issue?  Its not reproducible right now.

**Comment by alisey123** [comment]
I'm afraid I can't do the required but I will look around and try to find a solution. Thank you for your time. 
Regards!

**Comment by SiamKreative** [comment]
See related issue #289. Try to disable iThemes Security it and see if it works :)

**Comment by alisey123** [comment]
Unfortunately deactivating iTheme Security does not solve the problem.

**Comment by SiamKreative** [comment]
Did you check if your htaccess file still contains the problematic rule mentioned in #289 after deactivating iTheme Security?

**Comment by alisey123** [comment]
My htaccess file contains WordPress default rules. It does not have any extra lines. To make sure I deleted the file and let WordPress regenerate the htaccess file. so it does not include any extra rules! 
I've tried deactivating other plugins. Problem still exists.
this is page source after clicking on attachment links: `<body style="margin: 0px;"><img style="-webkit-user-select: none" src="http://mywebsite.com/?wpas-attachment=14330"></body>`

**Comment by elindydotcom** [comment]
We did some more research on this issue because a couple of other customers reported it as well.  An error is being thrown in the error log:  awesome-support/includes/file-uploader/class-file-uploader.php on line 253.  The error is "failed to open stream: connection refused."

Best guess is that the server is attempting to render the file using a local file path instead of via the url constructed in that area of the code.

Two workarounds that can be tried;

1. If running a  VM, add a mapping in the hosts file to point the ip of the vm to the domain.  
xxx.xxx.xxx.xxx  mydomain.com

2. Send all traffic through cloudlfare.

Both of those mitigations worked on the AWS server where we were able to reproduce this with a level of consistency.

**Comment by JanPaulKleijn** [comment]
Hi, I also have this problem and I think that this issue is caused by mostly the opposite of what elindydotcom said. The connection is refused _because_ the file is being opened with the url constructed in that area of the code.
The reason this does not work on a lot of servers is that on these servers **allow_url_fopen=0** is set. This is done for security reasons (read: [https://www.web24.com.au/tutorials/security-issues-allow_url_fopen](https://www.web24.com.au/tutorials/security-issues-allow_url_fopen)).
I also read that it is better to use cURL for these things, but that is up to Awesome Support.

**Comment by masrajha** [comment]
I use Awesome Support Plugin and i got this problem and i solved it. There are 2 ways:
1. Edit file includes/file-uploader/class-file-uploader.php
Go To Line 597 replace this code
`$link = add_query_arg( array( 'wpas-attachment' => $attachment['id'] ), home_url() );`
to
`$link = add_query_arg( array( 'attachment_id' => $attachment['id'] ), home_url() );`
this way is open attachment using default wordpress link

2. Open wordpress Dashboard. In Menu Ticket->Settings open Advanced Tab and Tic **Do Not Mask Attachment Links** option. Then Save Setting.
This way open attachment direct to file address.

**Comment by pwkip** [comment]
For me, switching the Render method from inline to download solved the problem. The setting is also under Tickets > Settings > Advanced > Attachment Overrides

**Comment by tednh** [comment]
This issue has been fixed on recently versions of Awesome Support Plugin. 
Regards
Ted
